envoy/network/address.h

```diff
@@ -238,6 +239,19 @@ class Instance {
   virtual const Network::SocketInterface& socketInterface() const PURE;
 };
 
+/*
+ * Used to store Instance in filter state.
+ */
+class InstanceConstSharedPtrAccessor : public Envoy::StreamInfo::FilterState::Object {
+public:
+  InstanceConstSharedPtrAccessor(InstanceConstSharedPtr ip) : ip_(std::move(ip)) {}
+
+  InstanceConstSharedPtr getIp() const { return ip_; }
```

Comment:
Return `const Instance&` or `OptRef<const Instance>` because we didn't want to share the ownership of the address.